### PR TITLE
Add date range picker options

### PR DESCRIPTION
**Pull request by Imundy**

### CODE REVIEW

File: components/date-picker/component.jsx [hunk 1]
```diff
@@ -39,8 +39,17 @@ function generateWeeks(month, dateFunctions) {
...
 		selectedDate: PropTypes.instanceOf(Date),
+		/** Sets the selected date range (use with asDateRange prop) */
+		selectedDateRange: PropTypes.shape({
+			start: PropTypes.instanceOf(Date),
+			end: PropTypes.instanceOf(Date),
+		}),
+		/** Returns a daate when selected. If asDateRangePicker is true, it will return a date range object matching the selectedDateRange prop shape */
 		setSelectedDate: PropTypes.func.isRequired,
+		/** specifies that the component should function as a date range picker */
+		asDateRangePicker: PropTypes.bool,
```
Comment (Imundy) [inline components/date-picker/component.jsx]: @dustinsoftware here are the API changes I mentioned. It should behave the same if the new props are left out (so non-breaking).

**Comment by Imundy** [comment]
Updated this with the date period picker as well.

File: components/date-period-picker/component.jsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import debounce from 'lodash.debounce';
+import { DatePicker, Input } from '../main';
+import { dateFunctionProps } from '../date-picker/date-function-props';
+import * as Styled from './styled';
+
+const DATE_FORMAT_STRING = 'M/d/yyyy';
+
+export class DatePeriodPicker extends PureComponent {
```
Comment (dustinsoftware) [inline components/date-period-picker/component.jsx]: Missing a jsdoc comment describing this component and its functionality

File: components/date-picker/component.jsx [hunk 3]
```diff
@@ -39,8 +39,18 @@ function generateWeeks(month, dateFunctions) {
...
 export class DatePicker extends Component {
 	static propTypes = {
+		/** Sets the selected date */
 		selectedDate: PropTypes.instanceOf(Date),
+		/** Sets the selected date range (use with asDateRange prop) */
+		selectedDateRange: PropTypes.shape({
+			start: PropTypes.instanceOf(Date),
+			end: PropTypes.instanceOf(Date),
+		}),
+		/** Returns a daate when selected. If asDateRangePicker is true, it will return a date range object matching the selectedDateRange prop shape */
```
Comment (dustinsoftware) [inline components/date-picker/component.jsx]: Typo: daate

**Comment by Imundy** [comment]
👍 Sounds good, once the beta packages are out I'll try to integrate it back into giving.

**Pull request merged**